### ISSUE TITLE
Pulumi Cloud `Summary` Tab shows drastically different details from CLI during `pulumi import`

### ISSUE THREAD

**Issue by phillipedwards** [pulumi/console-requests#339]
## Expected behavior
Pulumi Cloud `Summary` tab should show the same view as the CLI during operations.

During a `pulumi import` operation, Pulumi Cloud shows numerous changes are being made on the imported resources while the Pulumi CLI shows no changes are being made.

## Current behavior

The `Summary` tab in Pulumi Cloud shows numerous changes are being made to the imported resources, while the CLI shows no changes are being made.

## Steps to reproduce

1. Using the program below, create 2 stacks (a & b)
2. `pulumi up` stack a and retrieve the IDs of both the created VPC and Subnet
3. `pulumi stack select b` and add the IDs retrieved from step 2 to the appropriate `_import` properties in the ResourceOptions object.
4. `pulumi preview` on stack b
5. Navigate to Pulumi Cloud and see the proposed changes in the `Summary` tab are drastically different than what is reported from the CLI (no changes).

Sample program:
```python
"""An AWS Python Pulumi program"""

import pulumi
from pulumi_aws import ec2

vpc = ec2.Vpc(
    "my-vpc",
    cidr_block="10.100.0.0/24",
    enable_dns_hostnames=True,
    enable_dns_support=True,
    tags={
        "Owner": "phil",
    },
    # uncomment and replace with vpc id to import
    # opts=pulumi.ResourceOptions(
    #     import_="vpc-080cf4f34c24ee859",
    # ),
)

sub1 = ec2.Subnet(
    "sub1",
    vpc_id=vpc.id,
    cidr_block="10.100.0.0/28",
    tags={
        "Name": "sub1",
    },
    # uncomment and replace with subnet id to import
    # opts=pulumi.ResourceOptions(
    #     import_="subnet-0e326796a3cf1778b",
    # ),
)
```

CLI Output (preview of import)
![image](https://github.com/pulumi/pulumi-cloud-requests/assets/25461821/0c45378b-1e09-4e9d-8fe3-ab6270ca1dc2)

Pulumi Cloud Summary (preview of import)
![image](https://github.com/pulumi/pulumi-cloud-requests/assets/25461821/c7c2e3c9-4941-4f78-afd9-6ac34df98cff)
z

## Affected feature

Pulumi CLI version - 3.100.0

**Comment by phillipedwards** [comment]
cc @komalali this is the issue we chatted about yesterday.

**Comment by komalali** [comment]
Finally tracked down the issue here thanks to @pgavlin's help!

The issue is not with the service, but when the Pulumi engine serializes the event to send to the service. As such, it will need to be fixed in pulumi/pulumi and only be resolved using an updated version of the CLI (we can't make any mitigations in the service code that would fix this).

The issue is with the omitempty here: https://github.com/pulumi/pulumi/blob/2d0275ca777a0790b993b18b98998c9606f510a6/sdk/go/common/apitype/events.go#L142

When the detailedDiff object is empty, i.e. `{}`, it is dropped from the serialized json due to the `omitempty`. When it is rehydrated on the service, it is rehydrated as `nil` because it is not present in the json object. That causes [this line](https://github.com/pulumi/pulumi/blob/516979770f11d3a426239429731cf9f327c38836/pkg/backend/display/rows.go#L442) to be passed through and results in an input diff, which leads to the output we see in the cloud.

**Comment by phillipedwards** [comment]
@komalali You're amazing. Thank you for going the rabbit hole on this one and figuring it out!

**Comment by komalali** [comment]
The issue is closed because the bug is fixed in pu/pu but just wanted to drop a note here that this will be completely resolved for the end user when:
1. pulumi v3.103.0 is released (coming today)
2. Pulumi Cloud is updated using pulumi v3.103.0 (coming today, once 1 is complete)

The fix is only valid is using pulumi CLI v3.103.0 or greater.

I'll update this issue once all the pieces are in place.

**Comment by komalali** [comment]
> I'll update this issue once all the pieces are in place.

Took a little longer than expected but the pieces for this fix are all released now. You should not see this bug anymore if using pulumi CLI v3.103.0 or greater.
